### DISCUSSION TITLE
Is it possible to define an entity only using EntitySchema?

### DISCUSSION THREAD

**Discussion by rperfect** [mikro-orm/mikro-orm#2682] [Q&A]
Is it possible to define an entity only using the EntitySchema definition without having to also have the Interface or Class defined? 

I'd like to be able to define an entity at runtime purely from meta-data without any predefined classes or interfaces.

Something along the lines of what's on this page; https://mikro-orm.io/docs/entity-schema but without the "Book" interface.

I tried;
 
`
orm.getMetadata().set('Book', { entity meta data goes here });
`

...and quite a few other things, but it didn't work :-(

**Comment by B4nan** [comment]
> Something along the lines of what's on this page; mikro-orm.io/docs/entity-schema but without the "Book" interface.

Then just pass `any` instead of an interface. Or `Dictionary` or `Record<string, any>` or something like that. Dont try to work with the `MetadataStorage` directly, it needs `EntityMetadata` instances, and they need to be initialized (`sync` method needs to be called). Instead use `orm.discoverEntity()` - it is v5 only, but if you check the code, you can do the same things in v4 too, it's more of a syntax sugar (you might need to use some private methods in v4, but we are in JS, so easily doable).

https://github.com/mikro-orm/mikro-orm/blob/master/packages/core/src/MikroORM.ts#L123

The signature looks a bit wrong, you should be able to pass in the `EntitySchema` instance just fine.

```ts
const schema = new EntitySchema({ ... });
await orm.discoverEntity(schema);
```

Search existing issues for more details, iirc there was someone asking for the same thing.

**Reply by rperfect** [reply]
I upgraded to release 5.0.0-rc.0 and I was able to call the discoverEntity() method successfully with the following;

`await this.orm.discoverEntity(entitySchema as any);
`
But when I attempt to read an entity from the database it fails inside of the `EntityFactory.createEntity(data, meta, options)` method near the bottom because meta.definedProperties is undefined.

```
if (options.initialized && !entity.__gettersDefined) {
    Object.defineProperties(entity, meta.definedProperties);  // fails on this line; meta.definedProperties is undefined
}
```

The "data" and "meta" parameters into this method look pretty good and populated, it's just that meta doesn't have anything called "definedProperties" - it has both "properties" and "props" and they look good.

Is this just a 5.x dev issue?
